File: readropslibrary/src/main/java/com/readrops/readropslibrary/services/nextcloudnews/json/NextNewsUser.java
package com.readrops.readropslibrary.services.nextcloudnews.json;

public class NextNewsUser {

    private String userId;

    private String displayName;

    private long lastLoginTimestamp;

    private Avatar avatar;

    public NextNewsUser() {
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public long getLastLoginTimestamp() {
        return lastLoginTimestamp;
    }

    public void setLastLoginTimestamp(long lastLoginTimestamp) {
        this.lastLoginTimestamp = lastLoginTimestamp;
    }

    public Avatar getAvatar() {
        return avatar;
    }

    public void setAvatar(Avatar avatar) {
        this.avatar = avatar;
    }

    public class Avatar {

        private String data;

        private String mime;
    }

}

File: readropslibrary/src/main/java/com/readrops/readropslibrary/services/nextcloudnews/json/NextNewsUser.kt
package com.readrops.readropslibrary.services.nextcloudnews.json

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class NextNewsUser(val userId: String,
                        val displayName: String,
                        val lastLoginTimestamp: Long,
                        val avatar: Avatar?) {

    @JsonClass(generateAdapter = true)
    data class Avatar(val data: String,
                      val mime: String)
}